src/main/scala/tech/cryptonomic/conseil/tezos/PlatformDiscoveryOperations.scala

```diff
@@ -0,0 +1,77 @@
+package tech.cryptonomic.conseil.tezos
+
+import com.typesafe.config.Config
+import slick.ast.FieldSymbol
+import slick.jdbc.PostgresProfile
+import slick.jdbc.PostgresProfile.api._
+import tech.cryptonomic.conseil.tezos.PlatformDiscoveryTypes._
+import tech.cryptonomic.conseil.tezos.{TezosDatabaseOperations => TezosDb}
+import tech.cryptonomic.conseil.util.DatabaseUtil
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{ExecutionContext, Future}
+
+
+object PlatformDiscoveryOperations {
+
+  lazy val dbHandle: PostgresProfile.backend.Database = DatabaseUtil.db
+
+  private val tables = List(Tables.Blocks, Tables.Accounts, Tables.OperationGroups, Tables.Operations, Tables.Fees)
+  private val tablesMap = tables.map(table => table.baseTableRow.tableName -> table)
+
+  def getNetworks(config: Config): List[Network] = {
+    config.getObject("platforms").asScala.flatMap {
+      case (platform, strippedConf) =>
+        strippedConf.atKey(platform).getObject(platform).asScala.map {
+          case (network, _) =>
+            Network(network, network.capitalize, platform, network)
+        }.toList
+    }.toList
+  }
+
+  def getEntities(network: String)(implicit ec: ExecutionContext): Future[List[Entity]] = {
+    ApiOperations.countAll.map { counts =>
+      tablesMap.map(_._1).flatMap { tableName =>
+        counts.get(tableName).map { tableCount =>
+          Entity(
+            name = tableName,
+            displayName = makeDisplayName(tableName),
+            count = tableCount,
+            network = network
+          )
+        }.toList
+      }
+    }
+  }
+
+  private def makeDisplayName(name: String): String = {
+    name.capitalize.replace("_", " ")
+  }
+
+  def tableAttributes(tableName: String)(implicit ec: ExecutionContext): Future[List[Attributes]] = {
```

Comment:
Please validate the function parameter and either throw an exception or return an error type.

Reply:
Let's try to stick to avoiding exceptions and wrap into appropriate error types, even using cats ones, if needed.

We're trying to make all functions generally return a value, unless a critical failure happens

Reply:
What works best for you, I would say a failed `Future` as a first answer... but it might depend on where the parameter is passed-in and where the return value is handled

Reply:
I see usually there are returned just `Future[]` in other places so I thought `Future[]` will be fine here